app/models/proposal.rb

```diff
@@ -33,6 +33,7 @@ class Proposal < ActiveRecord::Base
   has_many :steps
   has_many :individual_steps, ->{ individual }, class_name: 'Steps::Individual'
   has_many :approvers, through: :individual_steps, source: :user
+  has_many :completers, through: :individual_steps, source: :completer
```

Comment:
Why not `step_approvers` then?

Right now we are in flux from a code world where there were `Approvals` and `Approvers` to a world of `Steps` and ...? what? `Steppers`?

I think architecturally it was absolutely correct to rename `approvals` to `steps` in order to reflect that a proposal evolves through various milestones in its workflow/lifetime, not all of which are "approval". Each of those steps gets assigned to someone initially. In the case of that "someone" being a generic mailbox (as all Tier 1 and Tier 2 NCR approvers are), there are then delegate users who actually perform the step. We want the audit trail (and the authorization delegation) that goes along with distinguishing between who was assigned the step and who actually performed the step. `StepPerformer`? `StepCompleter`? We landed on completer for the time being.

Naming things is the hardest thing in this job.

I feel strongly that the code should reflect, as much as possible, the realities of the business problems it is trying to solve. That helps ground the programmer in the business reality at the same time they are grokking the code itself.

How about we leave `completers` alone for the time being, until such time as we completely get rid of approvers. At that point perhaps a more perfect naming scheme will have presented itself. I like the simple poetic parity between `approver` and `completer`. In 66% of NCR steps, they are different Users, so it's not a trivial distinction.